src/powersoftau_export_json.js

```diff
@@ -44,43 +45,40 @@ export default async function exportJson(pTauFilename, verbose) {
 
     await fd.close();
 
-    return stringifyBigIntsWithField(curve.Fr, pTau);
-
-
+    return ffUtils.stringifyBigInts(pTau);
 
     async function exportSection(sectionId, groupName, nPoints, sectionName) {
-        const G = curve[groupName];
-        const sG = G.F.n8*2;
-
         const res = [];
+        const readFn = (groupName == "G1") ? readG1 : readG2;
         await binFileUtils.startReadUniqueSection(fd, sections, sectionId);
         for (let i=0; i< nPoints; i++) {
             if ((verbose)&&i&&(i%10000 == 0)) console.log(`${sectionName}: ` + i);
-            const buff = await fd.read(sG);
-            res.push(G.fromRprLEM(buff, 0));
+            res.push(await readFn(fd, curve, true));
         }
         await binFileUtils.endReadSection(fd);
 
         return res;
     }
 
     async function exportLagrange(sectionId, groupName, sectionName) {
-        const G = curve[groupName];
-        const sG = G.F.n8*2;
-
         const res = [];
+        const readFn = (groupName == "G1") ? readG1 : readG2;
+
+        // The lTauG1 section (#12) calculates powers up to power + 1,
+        // per preparePhase2 function in powersoftau_preparephase2.js
+        const lastPower = (sectionId == 12) ? power + 1 : power;
+
         await binFileUtils.startReadUniqueSection(fd, sections, sectionId);
-        for (let p=0; p<=power; p++) {
+        for (let p=0; p<=lastPower; p++) {
             if (verbose) console.log(`${sectionName}: Power: ${p}`);
             res[p] = [];
             const nPoints = (2 ** p);
             for (let i=0; i<nPoints; i++) {
                 if ((verbose)&&i&&(i%10000 == 0)) console.log(`${sectionName}: ${i}/${nPoints}`);
-                const buff = await fd.read(sG);
-                res[p].push(G.fromRprLEM(buff, 0));
+                res[p].push(await readFn(fd, curve, true));
             }
         }
-        await binFileUtils.endReadSection(fd, true);
+        await binFileUtils.endReadSection(fd);
```

Comment:
Why did you remove the skip boolean here?
```suggestion
        await binFileUtils.endReadSection(fd, true);
```

Reply:
The logic here should remove the need for the boolean:  https://github.com/iden3/snarkjs/pull/177/files#diff-8678b0d3cd8e708a369dbc86a4da23f29ca18e763f726c3cf1f8f624e1267075R65.

Basically, it looks like for some sections, the group values that are saved are up to the inputted (power + 1).  Hence the use of the `lastPower` variable above.

IMHO, using the boolean does slightly comprise the safety of the code, as it doesn't verify that it is reading all of the contents of the `ptau` file.

Reply:
Interesting. I'm fine with the change, especially for increased safety, but can you explain why `powersoftau_verify` uses the `true` argument at https://github.com/iden3/snarkjs/blob/master/src/powersoftau_verify.js#L455 and https://github.com/iden3/snarkjs/blob/master/src/powersoftau_verify.js#L485 ? I'm pretty sure we want the verify to be secure as possible, right?